util/dvsim/FpvCfg.py

```diff
@@ -0,0 +1,282 @@
+        # The email_summary will contain all the information from results_md
+        log.info("Create summary of FPV results")
+
+        results_str = "## " + self.results_title + " (Summary)\n\n"
+        results_str += "### " + self.timestamp_long + "\n\n"
+
+        colalign = ("center", ) * len(self.summary_header)
+        table = [self.summary_header]
+        for cfg in self.cfgs:
+            try:
+                table.append(cfg.result_summary[cfg.name])
+            except KeyError as e:
+                table.append([cfg.name, "ERROR", "N/A", "N/A", "N/A"])
+                log.error("cfg: %s could not find generated results_summary: %s", cfg.name, e)
+        if len(table) > 1:
+            self.results_summary_md = results_str + tabulate(
+                table, headers="firstrow", tablefmt="pipe", colalign=colalign)
+        else:
+            self.results_summary_md = results_str
+
+        log.info("[result summary]: %s", self.results_summary_md)
+
+        # Generate email results summary
+        colalign = ("left", ) + ("center", ) * (len(self.header) - 1)
+        email_table = [self.header]
+        error_message = ""
+
+        for cfg in self.cfgs:
+            email_result = cfg.result.get("fpv_summary")
+            if email_result is not None:
+                email_table.append([
+                    cfg.name,
+                    str(email_result["fpv_errors"]) + " E ",
+                    str(email_result["fpv_warnings"]) + " W ",
+                    str(email_result["proven"]) + " G ",
+                    str(email_result["cex"]) + " E ",
+                    str(email_result["undetermined"]) + " W ",
+                    str(email_result["covered"]) + " G ",
+                    str(email_result["unreachable"]) + " E ",
+                    email_result["pass_rate"],
+                    email_result["cov_rate"]
+                ])
+            messages = cfg.result.get("messages")
+            if messages is not None:
+                # TODO: temp disable printing out "fpv_warnings" in results_summary
+                # Will clean up FPV warnings first, then display "fpv_warnings"
+                error = self.parse_dict_to_str(messages, ["fpv_warnings"])
+                if error:
+                    error_message += "\n#### " + cfg.name + "\n"
+                    error_message += error
+
+        if len(email_table) > 1:
+            self.email_summary_md = results_str + tabulate(
+                email_table, headers="firstrow", tablefmt="pipe", colalign=colalign)
+        self.email_summary_md += error_message
+
+        return self.results_summary_md
+
+    def _gen_results(self):
+        # This function is called after the regression and looks for
+        # results.hjson file with aggregated results from the FPV run.
+        # The hjson file is required to follow this format:
+        # {
+        #   "messages": {
+        #      "errors"      : []
```

Comment:
do you intend to separate the `errors` and `warnings` from `fpv_errors` and `fpv_warnings`? if so, we should report the `errors` and `warnings` as well... otherwise, I would suggest to just combine these categories, similarly as this has been done for the lint flows: https://github.com/lowRISC/opentitan/blob/679cfce00ae2b1be938bb93d12bfde33b2de8505/util/dvsim/LintCfg.py#L106-L107 I.e., you would just be left with `errors` and `warnings`, which basically stand for tool flow error and warning messages.

Reply:
Initially was debating about it because I wanted to distinguish the Flow errors/warnings that are not from FPV, but I think it is not super necessary, users should be able to find the difference... Yeah i will just combine them so the output table won't be too large.